Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/MatchAndMergeWorkspacesTest.py

```diff
@@ -0,0 +1,74 @@
+
+    def setUp(self):
+        ws_list = []
+        for i in range(5):
+            ws_name = 'ws_' + str(i+1)
+            data_x = np.arange(i, (i+1)*10+0.1, 0.1)
+            data_y = np.arange(i, (i+1)*10, 0.1)
+            data_e = np.arange(i, (i+1)*10, 0.1)
+            CreateWorkspace(OutputWorkspace=ws_name, DataX=data_x, DataY=data_y, DataE=data_e)
+            ws_list.append(ws_name)
+        GroupWorkspaces(InputWorkspaces=ws_list, OutputWorkspace='ws_group')
+
+    def test_MatchAndMergeWorkspaces_executes(self):
+        x_min = np.array([0, 5, 10, 15, 20])
+        x_max = np.array([10, 20, 30, 40, 50])
+        ws_merged = MatchAndMergeWorkspaces(InputWorkspaces='ws_group', XMin=x_min, XMax=x_max)
+        self.assertIsInstance(ws_merged, MatrixWorkspace)
+        self.assertEqual(ws_merged.getNumberHistograms(), 1)
+        self.assertAlmostEqual(min(ws_merged.dataX(0)), 0, places=0)
+        self.assertAlmostEqual(max(ws_merged.dataX(0)), 50, places=0)
+
+    def test_MatchAndMergeWorkspaces_produces_correct_range(self):
+        x_min = np.array([2, 5, 10, 15, 20])
+        x_max = np.array([10, 20, 30, 40, 45])
+        ws_merged = MatchAndMergeWorkspaces(InputWorkspaces='ws_group', XMin=x_min, XMax=x_max)
+        self.assertIsInstance(ws_merged, MatrixWorkspace)
+        self.assertEqual(ws_merged.getNumberHistograms(), 1)
+        self.assertAlmostEqual(min(ws_merged.dataX(0)), 2, places=0)
+        self.assertAlmostEqual(max(ws_merged.dataX(0)), 45, places=0)
+
+    def test_MatchAndMergeWorkspaces_accepts_a_list_of_workspaces(self):
+        x_min = np.array([2, 5, 10])
+        x_max = np.array([10, 20, 30])
+        ws_group = AnalysisDataService.retrieve('ws_group')
+        ws_list = [ws_group[0], ws_group[1], ws_group[2]]
+        ws_merged = MatchAndMergeWorkspaces(InputWorkspaces=ws_list, XMin=x_min, XMax=x_max)
+        self.assertIsInstance(ws_merged, MatrixWorkspace)
+        self.assertEqual(ws_merged.getNumberHistograms(), 1)
+        self.assertAlmostEqual(min(ws_merged.dataX(0)), 2, places=0)
+        self.assertAlmostEqual(max(ws_merged.dataX(0)), 30, places=0)
+
+    def test_MatchAndMergeWorkspaces_accepts_a_mixture_of_ws_size(self):
+        x_min = np.array([2, 5, 10, 15, 20])
+        x_max = np.array([10, 20, 30, 40, 45])
+        ws_group = AnalysisDataService.retrieve('ws_group')
+        ConjoinWorkspaces(InputWorkspace1=ws_group[3],
+                          InputWorkspace2=ws_group[4],
+                          CheckOverlapping=False)
+        ws_list = [ws_group[0], ws_group[1], ws_group[2], ws_group[3]]
+        ws_merged = MatchAndMergeWorkspaces(InputWorkspaces=ws_list, XMin=x_min, XMax=x_max)
+        self.assertIsInstance(ws_merged, MatrixWorkspace)
+        self.assertEqual(ws_merged.getNumberHistograms(), 1)
+        self.assertAlmostEqual(min(ws_merged.dataX(0)), 2, places=0)
+        self.assertAlmostEqual(max(ws_merged.dataX(0)), 45, places=0)
+
```

Comment:
These are a good set of tests. In it is also generally a good idea, when unit testing, to test that things fail as expected too. For instance we have validation on the inputs so we could check that the algorithm fails when passed input that is known to be wrong and responds as we expect. [This](/mantidproject/mantid/blob/master/Framework/Crystal/test/PredictFractionalPeaksTest.h#L283) is an example in C++ but the aim is the same.